weather.py

```diff
@@ -0,0 +1,440 @@
+# Imports written in alphabetical order
+from os import path
+from csv_reader import CSVReader
+import sys
+from termcolor import colored
+import zipfile
+
+
+class Weatherman:  # Weatherman Application in one class
+
+    file_name_prefix = "Murree_weather_"  # The part of the file name that is same for all files
+
+    # List with months to access by numerical form
+    months = ["Jan", "Feb", "Mar",
+              "Apr", "May", "Jun",
+              "Jul", "Aug", "Sep",
+              "Oct", "Nov", "Dec"
+              ]
+
+    # Class Constructor
+    def __init__(self, path, highest_argument, average_argument, chart_form_argument, chart_form_combined):
+        self.__path = path
+        self.__highest_arguments = highest_argument
+        self.__average_arguments = average_argument
+        self.__chart_form_arguments = chart_form_argument
+        self.__chart_form_combined = chart_form_combined
+        self.files_path_prefix = self.__path + "/weatherfiles/" + Weatherman.file_name_prefix  # path common in all files
+
+        # Report Number to print
+        self.report_num = 1
+
+    def file_exists(self, date):
```

Comment:
This method does too much stuff, you are repeating this logic in many places. This should receive a file name and return `True` or `False` depending on the existence of the file.

Reply:
done